lib/protobuf/nats/client.rb

```diff
@@ -137,34 +193,40 @@ def nats_request_with_two_responses(subject, data, opts)
           timeout = opts[:timeout] || 60
 
           nats = ::Protobuf::Nats.client_nats_connection
-          inbox = nats.new_inbox
 
           # Publish to server
-          sub = nats.subscribe(inbox, :max => 2)
-          nats.publish(subject, data, inbox)
+          with_subscription do |sub_inbox|
+            nats.publish(subject, data, sub_inbox.inbox)
+
+            # Wait for reply
+            first_message = nats.next_message(sub_inbox.subscription, ack_timeout)
 
-          # Wait for reply
-          first_message = nats.next_message(sub, ack_timeout)
-          return :ack_timeout if first_message.nil?
-          first_message_data = first_message.data
-          return :nack if first_message_data == ::Protobuf::Nats::Messages::NACK
+            if first_message.nil?
+              nats.unsubscribe(sub_inbox.subscription)
```

Comment:
We'll want to remove the subscription from the connection pool after any timeout since it's a server _might_ send a response and potentially screw up the next request that checks out this inbox. We might actually want to write our own little connection_pool-ish gem since our use case is very different.